File: tests/Generator/resource/python/complex/reference_property_type.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union, Literal
from .property_type import PropertyType


# Represents a reference to a definition type
class ReferencePropertyType(PropertyType):
    type: Literal["reference"] = Field(alias="type")
    target: Optional[str] = Field(default=None, alias="target")
    template: Optional[Dict[str, str]] = Field(default=None, alias="template")
    pass

File: tests/Generator/resource/visualbasic/complex/ReferencePropertyType.vb
Imports System.Text.Json.Serialization

' Represents a reference to a definition type
Public Class ReferencePropertyType
    Inherits PropertyType
    <JsonPropertyName("target")>
    Public Property Target As Nullable(String)

    <JsonPropertyName("template")>
    Public Property Template As Nullable(Dictionary(Of String, String))

End Class

